src/rpc/rawtransaction.h

```diff
@@ -17,6 +17,6 @@ class Chain;
 UniValue SignTransaction(interfaces::Chain& chain, CMutableTransaction& mtx, const UniValue& prevTxs, CBasicKeyStore *keystore, bool tempKeystore, const UniValue& hashType);
 
 /** Create a transaction from univalue parameters */
-CMutableTransaction ConstructTransaction(const UniValue& inputs_in, const UniValue& outputs_in, const UniValue& locktime, const UniValue& rbf, const UniValue& assets_in);
+CMutableTransaction ConstructTransaction(const UniValue& inputs_in, const UniValue& outputs_in, const UniValue& locktime, const UniValue& rbf, const UniValue& assets_in, std::vector<CPubKey>* output_pubkeys_out = nullptr);
```

Comment:
can you document `output_pubkeys_out`? Looks like it's a flag for hack key in nonce :)

Reply:
Hm, it's possible that it could go away, actually. Originally, it was not so much a flag, as an alternative way to get the pubkeys from ConstructTransaction -- if it was non-null, the pubkeys would be put into output_pubkeys_out _instead_ of the nonces.

However, I ended up abandoning this mechanism (I think completely?) in favor of letting it stuff nonces, then extracting the nonces later. This is because I was unable to adequately control FundTransaction's behavior -- it was insisting on stuffing pubkeys into nonces when creating _change_, and I gave up on trying to dig deeply enough to understand how to prevent it.

I do think we should switch to something like this mechanism in the future, to get rid of the nonce-stuffing hack entirely. But I think PSBT needs to be full-featured and regular unsigned transactions deprecated first, so perhaps the world is not ready for this yet.